pkg/comp-functions/functions/common/netpol.go

```diff
@@ -9,29 +9,31 @@ import (
 )
 
 // CreateNetworkPolicy creates network policy in the instance namespace to allow other namespaces access to the service
-func CreateNetworkPolicy(sourceNs []string, instanceNs string, instance string, svc *runtime.ServiceRuntime) error {
+func CreateNetworkPolicy(sourceNs []string, instanceNs string, instance string, allowAll bool, svc *runtime.ServiceRuntime) error {
```

Comment:
We could change parameters so it takes a `Composite` interface type, then we can reduce the amount of params and make it easier to handle.

You could even specify a `SecurityGetter` interface which only contains the `GetSecurity()` function, to make it even more specific. Then you can add the `SecurityGetter` back to the `Composite` as an anonymous field, so every `Composite` is automaticalyl also a `SecurityGetter`.

Something like this:

```
type Composite interface {
	InfoGetter
	client.Object
        SecurityGetter
	SetInstanceNamespaceStatus()
	GetAllowedNamespaces() []string
}

type SecurityGetter interface {
       GetSecurity() vshnv1.Security
}

```

```suggestion
func CreateNetworkPolicy(source SecurityGetter, instanceNs string, instance string, svc *runtime.ServiceRuntime) error {
```

Reply:
The only reason I didn't use the interface it's because on some cases such as keycloak pg I pass the value as false

Reply:
We came to an agreement after PP.